docs/life_of_a_godwoken_transaction.md

```diff
@@ -0,0 +1,338 @@
+
+With enough background, the rule for executing backends can be introduced:
+
+* Using `to_id` in the layer 2 transaction, Godwoken locates the corresponding backend script
+* The backend script is executed to calculate the state after applying current transaction.
+
+Similar to account locks, the above rule is more of a conceptual rule. The actual backend script is only executed on-chain in a cancel challenge request.
+
+As examples, several different backends will be introduced here:
+
+#### sUDT
+
+sUDT is the most commonly used backend in Godwoken. All tokens, whether they are CKB or layer 1 sUDTs, will be represented as a layer 2 sUDT type in Godwoken. A layer 2 sUDT backend script is defined via the following specification:
+
+* `code_hash` and `hash_type` are pre-determined by each Godwoken deployment
+* `args` contains the script hash of layer 1 sUDT type script. In case CKB is used, `args` will be `0x0000000000000000000000000000000000000000000000000000000000000000`.
+
+This way we can correlate layer 1 sUDT type script, with its corresponding layer 2 sUDT backend script.
+
+The layer 2 sUDT backend provides [ERC20](https://eips.ethereum.org/EIPS/eip-20) compatible interface, even though you are not using Solidity and EVM, a similar API will be available for you to use.
+
+Notice sUDT contract account is typically created by Godwoken automatically when processing deposition cells. One don't typically create new contract account using sUDT as backend script.
+
+An implementation for the layer 2 sUDT backend can be found [here](https://github.com/nervosnetwork/godwoken/blob/v0.1.0/c/contracts/sudt.c).
+
+#### MetaContract
+
+MetaContract is a special backend in Godwoken:
+
+* It is reserved as account ID 0
+* You cannot create a new contract account using MetaContract as the backend
+
+The sole purpose of MetaContract now, is to create contract account given a particular backend.
+
+**TODO**: create polyjuice root account
+
+#### Polyjuice
+
+Polyjuice is the main backend we use now in Godwoken. It allows us to create a contract account using EVM bytecode, the resulting account, will be able to execute smart contracts written for Ethereum. Polyjuice aims at 100% compatibility in EVM level, meaning all programs runnable on Ethereum, can be run on Godwoken powered by Polyjuice.
+
+For more details on polyjuice, please refer to Life of a Polyjuice Transaction(**TODO**: add link when the document is ready).
+
+### Managing Account Locks & Backends
+
+While one is free to create as many accounts as possible, Godwoken now only supports whitelisted account locks & backends for a security reasons. The list now include:
```

Comment:
Is 'whitelisted' accurate here? Developers can deploy a new Godwoken instance with different account locks anyway, there is no way to really 'whitelist' account locks.

Reply:
Whitelist is accurate here, here it means `whitelist` the actual script in the account lock, not the full account lock. While one is free to create new account locks, the underlying script used in account lock is whitelisted now.